### PR TITLE
sys: disable jit on apple aarch64

### PR DESCRIPTION
**Pull request by b8591340**
clang fails with
```
    Undefined symbols for architecture arm64:
      "___clear_cache", referenced from:
          _sljit_generate_code in libforeign.a(pcre2_jit_compile.o)
    ld: symbol(s) not found for architecture arm64
```

I’ve only tested an `aarch64-apple-ios` build, but see [here](https://bugreports.qt.io/browse/QTBUG-62993?gerritReviewStatus=All) and [here](https://github.com/Homebrew/homebrew-core/pull/57419) for the others.

I also disabled it on the equivalent x86_64 targets to keep an identical binary across simulators and Catalyst (`x86_64-apple-ios-macabi`); the latter might be debatable assuming the symbols are there as they should be, and we might want to enable it there.

### CODE REVIEW

File: pcre2-sys/build.rs [hunk 1]
```diff
@@ -93,9 +93,13 @@ fn main() {
         .define("PARENS_NEST_LIMIT", "250")
         .define("PCRE2_STATIC", "1")
         .define("STDC_HEADERS", "1")
-        .define("SUPPORT_JIT", "1")
         .define("SUPPORT_PCRE2_8", "1")
         .define("SUPPORT_UNICODE", "1");
+    if !target.starts_with("aarch64-apple") &&
```
Comment (BurntSushi) [inline pcre2-sys/build.rs]: Could you add a comment before this `if` block explaining why we don't enable the JIT in these cases?

I might prefer defining a function, `fn enable_jit(builder: &mut cc::Build)` instead, and documenting that as appropriate.

**Review by BurntSushi — CHANGES_REQUESTED** [review]
Seems fine to me! But would like to leave a trail of breadcrumbs in the code so that it's easy to tell when or if we should re-enable it again.

File: pcre2-sys/build.rs [hunk 2]
```diff
@@ -151,3 +152,29 @@ fn pcre2_sys_static() -> Option<bool> {
...
+// aarch64-apple-darwin       https://github.com/Homebrew/homebrew-core/pull/57419
+// x86_64-apple-ios           disabled for device–simulator consistency (not tested)
+// x86_64-apple-tvos          disabled for device–simulator consistency (not tested)
+// armv7-apple-ios            assumed equivalent to aarch64-apple-ios (not tested)
+// armv7s-apple-ios           assumed equivalent to aarch64-apple-ios (not tested)
+// i386-apple-ios             assumed equivalent to aarch64-apple-ios (not tested)
+// x86_64-apple-ios-macabi    disabled out of caution (not tested) (needs attention)
+//
+// We may want to monitor developments on the `aarch64-apple-darwin` front as they may end up
+// propagating to all `aarch64`-based targets and the `x86_64` equivalents.
```
Comment (BurntSushi) [inline pcre2-sys/build.rs:173]: Lovely, thanks so much! :D

**Review by BurntSushi — APPROVED** [review]

**Comment by BurntSushi** [comment]
This PR is on crates.io in `pcre2-sys 0.2.5`. (I've also updated ripgrep master to use the new version of `pcre2-sys`.)

Thanks again!

**Comment by carenas** [comment]
the following [patch](/carenas/sljit/commit/a9876e5728903103f18c8c1fbc37101cfeba1125) might help, but it is likely incomplete as the resulting code is known to cause older versions of iOS to crash (unless the iDevice was rooted), you might be interested on testing zherczeg/sljit#90 if the objective is to have JIT enabled in Apple Silicon eventually

**Comment by BurntSushi** [comment]
@carenas Thanks. Yes, it would be nice to have the JIT enabled on Apple silicon eventually. But not only do I not have access to Apple's new silicon, but I won't have the bandwidth to do that kind of testing anyway. Thank you for the heads up though!

**Pull request merged**